### PR TITLE
Feat/vscode web extension

### PR DESCRIPTION
**Pull request by brian-mulier-p**

### CODE REVIEW

File: src/web/kestraFsProvider.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+		return (await (await this.callFlowsApi(`/${this.namespace}/${flowId}?source=true`)).json()).source;
+	}
+
+	async stat(uri: vscode.Uri): Promise<vscode.FileStat> {
+		if (this.isExcludedFolder(uri)) {
+			throw vscode.FileSystemError.FileNotFound(uri);
+		}
+
+		if (this.isFlowsDirectory(uri)) {
+			const defaultDirectorySize = 4096;
```
Comment (Skraye) [inline src/web/kestraFsProvider.ts]: Create a const

File: src/web/kestraFsProvider.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+	private extractFlowId(uri: vscode.Uri): string {
+		return uri.path.substring(uri.path.lastIndexOf("/") + 1);
+	}
+
+	private async getFlowSource(uri: vscode.Uri): Promise<string> {
+		const flowId = this.extractFlowId(uri);
+		return (await (await this.callFlowsApi(`/${this.namespace}/${flowId}?source=true`)).json()).source;
+	}
+
+	async stat(uri: vscode.Uri): Promise<vscode.FileStat> {
```
Comment (Skraye) [inline src/web/kestraFsProvider.ts:90]: Those are more metadata instead of Stats ?

**Review by Skraye — CHANGES_REQUESTED** [review]
Not much, just the thing above using stats instead of Meta can be confusing

File: src/web/kestraFsProvider.ts (hunk 2 above)
Comment (brian-mulier-p) [reply src/web/kestraFsProvider.ts:90]: That's the FileProviderSystem interface that brings this :P

Comment (Skraye) [reply src/web/kestraFsProvider.ts:90]: Wasn't sure, but you confirm it thanks

**Pull request merged**